### ISSUE TITLE
0.9.5 broke Enum null values provided as arguments

### ISSUE THREAD

**Issue by exogen** [graphql/graphql-js#840]
A [GraphBrainz](https://github.com/exogen/graphbrainz) tests started failing with v0.9.5 and above. I have an enum like:

```js
export const CoverArtImageSize = new GraphQLEnumType({
  name: 'CoverArtImageSize',
  description: `The image sizes that may be requested at the [Cover Art
Archive](https://musicbrainz.org/doc/Cover_Art_Archive).`,
  values: {
    SMALL: {
      name: 'Small',
      description: 'A maximum dimension of 250px.',
      value: 250
    },
    LARGE: {
      name: 'Large',
      description: 'A maximum dimension of 500px.',
      value: 500
    },
    FULL: {
      name: 'Full',
      description: 'The image’s original dimensions, with no maximum.',
      value: null
    }
  }
})
```

(Yes, I had this `null` valued Enum even before it was 'officially' supported, and simply worked around it by checking for `'FULL'` in the resolver.)

As of v0.9.5, I'm no longer able to supply `FULL` as an argument, as in this query:

```graphql
  {
    lookup {
      release(mbid: "b84ee12a-09ef-421b-82de-0441a926375b") {
        coverArt {
          front(size: LARGE)
          back(size: SMALL)
          fullFront: front(size: FULL)
        }
      }
    }
  }
```

This now results in an error from the server:

```
GraphQLError: Argument "size" has invalid value FULL.
Expected type "CoverArtImageSize", found FULL.
```

But clearly `FULL` is a valid value. It seems the new `null` support from #836 didn't extend support to arguments and resulted in breaking what was at least workaround-able.

(I can see how this could indeed pose problems for people when determining "no argument provided" vs. "null/undefined enum value provided", but that seems solvable by checking for `'arg' in args` and making sure `graphql-js` only populates passed-in args.)

**Comment by exogen** [comment]
(I guess a more accurate title would be that 0.9.5 claimed to add support for `null` valued Enums, but it's not actually full support. I realize that `'FULL'` and not `null` was actually being passed as the value in my code before.)

**Comment by exogen** [comment]
I can work on a PR for this if some details are fleshed out. @leebyron When an arg isn't provided, should its value be `undefined`, treated distinctly from passing an enum whose value is `null`?

**Comment by wincent** [comment]
If I understand your issue correctly, seems like there's an oversight here.  Your query is indeed valid and should be accepted.

**Comment by exogen** [comment]
@wincent I agree! I'm working on a fix for this and have narrowed it down to the [`isNullish` check in `isValidLiteralValue`](https://github.com/graphql/graphql-js/blob/7287c7ccecc58c97a3745a2db64f1bd5ae1ea7a2/src/utilities/isValidLiteralValue.js#L120-L121), which wasn't updated to account for null-valued Enums.
